File: src/Client.Desktop/Extensions/SpriteBatchExtensions.cs
using Client.View;

namespace Microsoft.Xna.Framework.Graphics
{
    public static class SpriteBatchExtensions
    {
        public static void Draw(this SpriteBatch spriteBatch, Vector2 position, TextureHolder texture)
        {
            spriteBatch.Draw(texture.Texture2D, position, texture.Source, Color.White);
        }
    }
}

File: src/v2/core/src/com/mygdx/game/extensions/SpriteBatchExtensions.java
package com.mygdx.game.extensions;


import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.mygdx.game.view.Vector2;
import com.mygdx.game.view.TextureHolder;

public class SpriteBatchExtensions {

    public static void Draw(SpriteBatch spriteBatch, Vector2 position, TextureHolder texture) {
        spriteBatch.draw(texture.Texture2D, position.X(), position.Y(), texture.Source.width, texture.Source.height);
    }
}
